### PR TITLE
arash/Shared components

### PR DESCRIPTION
**Pull request by arash-fs**
# Description

shared components between `DBot, DMT5 and DTrader`

Changes:

-   fixes issue with item
-   ...

## Type of change

-   [ ] Bug fix
-   [ ] New feature
-   [x] Refactor code
-   [ ] Update translation text
-   [ ] Breaking change
-   [ ] Dependency update
-   [ ] This change requires a documentation update
-   [ ] Release

### CODE REVIEW

File: src/components/custom/_dhero.js [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    width: 100%;
+    padding: 10rem 0;
+    display: flex;
+    flex-direction: column;
+`
+const TryForFree = styled(Button)`
+    border: 2px solid var(--color-red);
+    font-weight: bold;
+    line-height: 1.43;
+    margin-top: 3.3rem;
```
Comment (oskar-binary) [inline src/components/custom/_dhero.js]: Why is this needed? (except the margin) doesn't the Join us for free button have the same styling as a normal button?

File: src/components/custom/_dhero.js [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    height: 100%;
+    max-width: 584px;
+    max-height: 375px;
+`
+const GoToLiveDemo = styled(Button)`
+    border: 2px solid var(--color-black-5);
+    font-weight: bold;
+    line-height: 1.43;
+    margin: 3.3rem 0 0 2.4rem;
+    background-color: var(--color-black);
```
Comment (oskar-binary) [inline src/components/custom/_dhero.js]: can't we use the button sedcondary for this? (except for the margin)

File: src/components/custom/_dtrading.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    flex-direction: ${props => props.flex_direction};
+    width: 100%;
+    display: flex;
+    margin-top: 4rem;
+
+    @media ${device.tabletL} {
+        flex-direction: column;
+    }
+`
+const DTrading = ({ trading, reverse, two_title }) => {
```
Comment (oskar-binary) [inline src/components/custom/_dtrading.js]: Why not smaller layout components instead of one big component for the whole thing?

File: src/components/custom/_dhero.js (hunk 1 above)
Comment (arash-fs) [reply src/components/custom/_dhero.js]: fixed

File: src/components/custom/_dhero.js (hunk 2 above)
Comment (arash-fs) [reply src/components/custom/_dhero.js]: fixed

File: src/components/custom/_dtrading.js (hunk 3 above)
Comment (arash-fs) [reply src/components/custom/_dtrading.js]: because I think it's better to have anything together. also, this is not a very big component. it just handles the trading section with some options that helps to handle DT5 too. because DMT5 is a little bit different.

File: src/pages/dbot/index.js [hunk 4]
```diff
@@ -1,28 +1,121 @@
...
+            translate_text='Maximise profits,<0/>limit losses'
+            components={[
+                <br key={0} />
+            ]}
+        />,
+        subtitle: localize(
+            'Use analysis tools and indicators and include take-profit, stop-loss and other smart logic to maximise your profits, and limit losses.',
+        ),
+        image_name: 'dbot-maximise-profits.png',
+        image_alt: 'Analysis tool',
```
Comment (oskar-binary) [inline src/pages/dbot/index.js]: and here

File: src/pages/dbot/index.js [hunk 5]
```diff
@@ -1,28 +1,121 @@
...
+            translate_text='Track your<0/>performance'
+            components={[
+                <br key={0} />
+            ]}
+        />,
+        subtitle: localize(
+            'See how your bot is performing as it executes each trade and receive notifications via Telegram.',
+        ),
+        image_name: 'dbot-track-your-performance.png',
+        image_alt: 'Track performace',
```
Comment (oskar-binary) [inline src/pages/dbot/index.js]: and here

File: src/pages/dbot/index.js [hunk 6]
```diff
@@ -1,28 +1,121 @@
...
+            translate_text='Get<0/>integrated help'
+            components={[
+                <br key={0} />
+            ]}
+        />,
+        subtitle: localize(
+            'Access tutorials, guides, and reference information as you build your bot.',
+        ),
+        image_name: 'dbot-get-integrated-help.png',
+        image_alt: 'Integrated help',
```
Comment (oskar-binary) [inline src/pages/dbot/index.js]: and here

File: src/pages/dbot/index.js [hunk 7]
```diff
@@ -1,28 +1,121 @@
...
+            translate_text='Save<0/>your strategies'
+            components={[
+                <br key={0} />
+            ]}
+        />,
+        subtitle: localize(
+            'Enjoy the convenience and security of storing your strategies on your Google Drive.',
+        ),
+        image_name: 'dbot-save-your-strategies.png',
+        image_alt: 'Save bot strategy',
```
Comment (oskar-binary) [inline src/pages/dbot/index.js]: and here

File: src/components/custom/_dhero.js [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+        window.open(deriv_app_url, '_blank')
+    }
+    const DLogo = styled(Logo)`
+        margin-right: 1.6rem;
+    `
+    return (
+        <Wrapper>
+            <StyledContainer>
+                <div>
+                    <StyledHeader as="h4" weight={500}>
```
Comment (oskar-binary) [inline src/components/custom/_dhero.js]: SEO request, can we make this into h1

File: src/components/custom/_dhero.js [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+                        <DLogo />
+                        {title}
+                    </StyledHeader>
+                    <HeroContent>
+                        <StyledContent as="h1">
```
Comment (oskar-binary) [inline src/components/custom/_dhero.js]: and this into h2

File: src/pages/dbot/index.js [hunk 10]
```diff
@@ -1,28 +1,121 @@
 import React, { Component } from 'react'
 import styled from 'styled-components'
-import { Tutorial } from './_tutorial'
-import { TradeWithDbot } from './_trade-with-dbot'
-import AutomateTrading from './_automate-trading'
-import Banner from './_banner'
-import HowItWorks from './_how-it-works'
+import DBotAnimation from './dbot-animation.json'
```
Comment (oskar-binary) [inline src/pages/dbot/index.js]: please add the lotties in `src/images/lotties` folder - the dbot, dtrader, dmt5 lotties will be used on the homepage too

File: src/pages/dbot/index.js [hunk 11]
```diff
@@ -1,28 +1,121 @@
...
+            translate_text='Start with a popular<0/>strategy'
+            components={[
+                <br key={0} />
+            ]}
+        />,
+        subtitle: localize(
+            'Martingale, D’Alembert, Oscar’s Grid, Cuttler’s RSI, Bollinger Bands, and SMA Crossover - load and customise proven strategies or create your own from scratch.',
+        ),
+        image_name: 'dbot-strategy.png',
+        image_alt: 'Strategy',
```
[the remark this answers is not in the document]
Comment (arash-fs) [reply src/pages/dbot/index.js]: done

File: src/pages/dbot/index.js [hunk 12]
```diff
@@ -1,28 +1,121 @@
...
+            translate_text='Build your strategy<0/>visually'
+            components={[
+                <br key={0} />
+            ]}
+        />,
+        subtitle: localize(
+            'Simply drag, drop, and configure pre-built blocks and indicators onto a canvas to build your bot. No coding needed.',
+        ),
+        image_name: 'dbot-build-strategy.png',
+        image_alt: 'Build bot strategy',
```
[the remark this answers is not in the document]
Comment (arash-fs) [reply src/pages/dbot/index.js]: done

File: src/pages/dbot/index.js (hunk 4 above)
Comment (arash-fs) [reply src/pages/dbot/index.js]: done

File: src/pages/dbot/index.js (hunk 5 above)
Comment (arash-fs) [reply src/pages/dbot/index.js]: done

File: src/pages/dbot/index.js (hunk 6 above)
Comment (arash-fs) [reply src/pages/dbot/index.js]: done

File: src/pages/dbot/index.js (hunk 7 above)
Comment (arash-fs) [reply src/pages/dbot/index.js]: done

File: src/components/custom/_dhero.js (hunk 8 above)
Comment (arash-fs) [reply src/components/custom/_dhero.js]: this is 24px in the design (platform name) we can't use h1 here

File: src/components/custom/_dhero.js (hunk 9 above)
Comment (arash-fs) [reply src/components/custom/_dhero.js]: the same reason here

File: src/pages/dbot/index.js (hunk 10 above)
Comment (arash-fs) [reply src/pages/dbot/index.js]: done

File: src/components/custom/_dhero.js (hunk 8 above)
Comment (mustofa-binary) [reply src/components/custom/_dhero.js]: we can use h1 and override the font size. its bcause in marketing point of view, seo will find h1 first, and the h1 is `dtrader` `dbot` `dmt5` that we want to highlight, we have to change our mindset for this to match with seo perspective :pray:

File: src/components/custom/_dhero.js [hunk 13]
```diff
@@ -0,0 +1,136 @@
...
+                            {content}
+                        </StyledContent>
+                    </HeroContent>
+                    {join_us_for_free ? (<TryForFree secondary onClick={toggleModal}>
```
Comment (mustofa-binary) [inline src/components/custom/_dhero.js]: `join_us_for_free && (<TryForFree secondary onClick={toggleModal}>{localize('Join us for free')}</TryForFree>)`

same goes for below two

File: src/components/custom/_dhow-it-works.js [hunk 14]
```diff
@@ -1,38 +1,39 @@
...
 import device from 'themes/device.js'
 
-const StyledSection = styled(SectionContainer)`
-    background-image: linear-gradient(to bottom, var(--color-grey-6), var(--color-white));
-`
-
 const StyledHeader = styled(Header)`
     margin-top: ${props => (props.as === 'h2' ? '0.8rem' : '0')};
-
+    ${props => props.font_size ? ('font-size:' + props.font_size) : ('')};
```
Comment (mustofa-binary) [inline src/components/custom/_dhow-it-works.js]:
```suggestion
    ${props => props.font_size && ('font-size:' + props.font_size)};
```

File: src/components/custom/_dtrading.js [hunk 15]
```diff
@@ -0,0 +1,104 @@
...
+    return (
+        <StyledSection>
+            <StyledContainer>
+                {trading.map((item, index) => {
+                    let is_even = reverse ? (index + 1) % 2 === 0 : index % 2 === 0
```
Comment (mustofa-binary) [inline src/components/custom/_dtrading.js]: why do we need to pass in reverse props?
everything modulus 2 will be either 1 or 0, so you can use `index % 2` only no need to do `index % 2 === 0`

File: src/components/elements/dropdown.js [hunk 16]
```diff
@@ -99,7 +99,7 @@ const Arrow = styled(Chevron)`
     right: 8px;
     top: 25%;
     transition: transform 0.2s linear;
-    ${props => (props.expanded ? 'transform: rotate(-180deg);' : '')}
+    ${props => (props.expanded ? '' : 'transform: rotate(-180deg);')}
```
Comment (mustofa-binary) [inline src/components/elements/dropdown.js]:
```suggestion
    ${props => (props.expanded && 'transform: rotate(-180deg);')}
```

File: src/pages/dbot/_dbot-video.js [hunk 17]
```diff
@@ -0,0 +1,160 @@
...
+            <Container>
+                <TabsWrapper>
+                    <Tab>
+                        <Step
+                            as="h4"
+                            lh="1.5"
+                            align="left"
+                            no_margin
+                            image_name='dbot-step1.png'
+                            current_step_image={this.state.current_step_image}
```
Comment (mustofa-binary) [inline src/pages/dbot/_dbot-video.js]: we can actually use state like `step: 1`
and in the current_step_image={`dbot-step${this.state.step}.png`}

File: src/pages/dtrader/index.js [hunk 18]
```diff
@@ -1,30 +1,85 @@
...
 `
 const Dtrader = () => {
     return (
         <Layout>
-            <SEO title={localize('Dtrader')} />
-            <Hero />
+            <SEO title={localize('DTrader')} />
+            <DHero
+                title={localize('DTrader')}
+                content={<Localize translate_text="Everything you <0/>need<0/>and more" components={[<br key={0} />]} />}
```
Comment (mustofa-binary) [inline src/pages/dtrader/index.js]: the width of the title is the same, thats why the text will wrap by itself, we dont need to manyally `br` the text, just use the specific width from the zeplin in the DHero component title

File: src/components/custom/_dhero.js (hunk 8 above)
Comment (arash-fs) [reply src/components/custom/_dhero.js]: done

File: src/components/custom/_dhero.js (hunk 9 above)
Comment (arash-fs) [reply src/components/custom/_dhero.js]: changed to h1

File: src/components/custom/_dhow-it-works.js (hunk 14 above)
Comment (arash-fs) [reply src/components/custom/_dhow-it-works.js]: done

File: src/components/custom/_dtrading.js (hunk 15 above)
Comment (arash-fs) [reply src/components/custom/_dtrading.js]: done

File: src/components/elements/dropdown.js (hunk 16 above)
Comment (arash-fs) [reply src/components/elements/dropdown.js]: we should pass a string and compare it with the string value, based on styled component.
